core/io/file_access.cpp

```diff
@@ -97,6 +97,10 @@ Ref<FileAccess> FileAccess::open(const String &p_path, int p_mode_flags, Error *
 			if (r_error) {
 				*r_error = OK;
 			}
+			if (ret.is_null()) {
+				// We need to agree on whether ret.is_null() means 'file cannot be opened' or 'file not found'
+				*r_error = ERR_FILE_NOT_FOUND;
+			}
```

Comment:
```suggestion
```
This can never happen, as this section only happens if `ret.is_valid()` is `true` which means `ret.is_null()` is `false`. Further you would have to check that `r_error` is not null, in which case you would get a segmentation fault here.